source/fab/steps/analyse.py

```diff
@@ -437,3 +431,22 @@ def _add_unreferenced_deps(self, symbols: Dict[str, Path],
             # add the file and it's file deps
             sub_tree = extract_sub_tree(source_tree=all_analysed_files, root=analysed_fpath)
             build_tree.update(sub_tree)
+
+
+def _load_analysis_file(fpath) -> Dict[Path, AnalysedFile]:
+    # Return the contents of an analysis csv.
+    results: Dict[Path, AnalysedFile] = dict()
+    try:
+        with open(fpath, "rt") as csv_file:
+            dict_reader = csv.DictReader(csv_file)
+
+            # read every row and convert into an AnalysedFile object
+            for row in dict_reader:
+                analysed_file = AnalysedFile.from_str_dict(row)
+                results[analysed_file.fpath] = AnalysedFile.from_str_dict(row)
```

Comment:
Isn't an `AnalysedFile` object being created twice for the same row? Should the `analysed_file` object be used instead of creating a new object?

Reply:
Yes, sorry. I suspect this might have been JetBrains' magic wand, which we've been using with glee to resolve conflicts where possible. I recently noticed, for the first time ever, that it didn't behave as expected so we'll keep a closer eye on this from now on, and maybe do it old skool instead.